units/provisioning_setup_host.go

```diff
@@ -833,3 +839,53 @@ func parseLsblkOutput(lsblkOutput string) ([]blockDevice, error) {
 
 	return devices.BlockDevices, nil
 }
+
+// GetGithubTokensForTask returns a read-only token for owner/repo associated with the task
+// and a read-only token for each module associated with the task.
+func GetGithubTokensForTask(taskId string, ctx context.Context) (string, []string) {
+	// Do not error at any point while trying to populate github tokens
+	// because if the repo is not private, cloning the repo will still work.
+	// Either way, we should still spin up the host even if we can't
+	// fetch the data.
+	var projectOwner, projectRepo string
+	var modules map[string]*manifest.Module
+	t, err := task.FindOneId(taskId)
+	if err != nil && t != nil {
+		mfest, err := manifest.FindFromVersion(t.Version, t.Project, t.Revision, t.Requester)
+		if err != nil {
+			modules = mfest.Modules
+		}
+		p, err := model.FindMergedProjectRef(t.Project, t.Version, false)
+		if err != nil {
+			projectOwner = p.Owner
+			projectRepo = p.Repo
+		}
+	}
+	var githubAppToken string
+	var moduleTokens []string
+	settings, err := evergreen.GetConfig(ctx)
+	opts := &github.InstallationTokenOptions{
+		Permissions: &github.InstallationPermissions{
+			Contents: utility.ToStringPtr(thirdparty.GithubPermissionRead),
+		},
+	}
+
+	if projectOwner != "" && projectRepo != "" && err == nil {
+		// Ignore any errors because if the repo is not private, cloning the repo will still work.
+		// Either way, we should still spin up the host even if we can't fetch the data.
+		githubAppToken, _ = settings.CreateGitHubAppAuth().CreateInstallationToken(ctx, projectOwner, projectRepo, opts)
+
+	}
+
+	for moduleName, module := range modules {
+		if module.Repo == "" {
+			_, module.Repo, _ = thirdparty.ParseGitUrl(module.URL)
```

Comment:
tiny nit: rather than setting the module's repo, would be a little cleaner to just define a local `var repo string` above, set it to `module.Repo`, and then overwrite it with the parsed result if `module.Repo` is empty.